recipe/patch_yaml/scipy.yaml

```diff
@@ -1,3 +1,28 @@
+# missing lower bounds for numpy after 2.0 migration, c.f.
+# https://github.com/conda-forge/numpy-feedstock/issues/324
+# the upper bound has been there correctly, c.f.
+# https://github.com/conda-forge/scipy-feedstock/commit/0711eb22f502ce2f1ea1bd7a22ffd4b8f73b73a7
+if:
+  name: scipy
+  version: 1.14.0
+
+then:
+  # https://github.com/scipy/scipy/blob/v1.14.0/pyproject.toml#L55
+  - add_constrains: numpy >=1.23.5
+
+---
+
+if:
+  name: scipy
+  version_lt: 1.14.0
+  version_ge: 1.13.0
+
+then:
+  # https://github.com/scipy/scipy/blob/v1.13.1/pyproject.toml#L55
+  - add_constrains: numpy >=1.22.4
```

Comment:
Am probably forgetting something, but is there a reason we use `add_constrains` instead of `add_depends`?

Since `numpy` is always a dependency of `scipy`, the latter is just restricting it more when solving and not adding a new dependency (admittedly so is `add_constrains` so the differences may only show up in the subtleties of how & when different solvers apply the two)

Reply:
I think they're functionally equivalent in this case? 🤷 

Happy to go either way. I guess using `replace_depends` is slightly cleaner, so that's what I've pushed now. Will update the diff as soon as the script finishes.

Reply:
Agreed. Hence non-blocking

Like that even better 👍